### PR TITLE
Feat(frontend): add  german translation

### PR DESCRIPTION
**Pull request by MariePrandstetter**
#4171

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

**Comment by lebouquetin** [comment]
Is it possible to include this in Tracim 3.8?

File: frontend/i18next.scanner/de/translation.json [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+  "Error while saving the comment": "Fehler beim Speichern des Kommentars",
+  "Show in publications": "In Publikationen anzeigen",
+  "Open_action": "Öffnen",
+  "Download {{filename}}": "Herunterladen {{filename}}",
+  "Sign in": "Anmelden",
+  "Login:": "Anmelden:",
+  "Password:": "Passwort:",
+  "Comment_noun": "kommentieren",
+  "New publication": "Neue Publikation",
+  "New content": "New content",
```
Comment (gbonaspetti) [inline frontend/i18next.scanner/de/translation.json:369]:
```suggestion
  "New content": "Neuer Inhalt",
```
?

File: frontend/i18next.scanner/de/translation.json [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+  "Description": "Beschreibung",
+  "Name": "Name",
+  "Filters": "Filtern",
+  "Close": "Schließen",
+  "Only search in": "Nur suchen in",
+  "Hide {{filter}}": "Ausblenden {{filter}}",
+  "Creation": "Erstellung",
+  "Show {{filter}}": "Anzeigen {{filter}}",
+  "Last Intervention": "Letzte Intervention",
+  "Loading recent activities…": "Laden der letzten Aktivitäten...",
```
Comment (gbonaspetti) [inline frontend/i18next.scanner/de/translation.json:420]:
```suggestion
  "Loading recent activities…": "Laden der letzten Aktivitäten…",
```

File: frontend_lib/i18next.scanner/de/translation.json [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  "The type of this file is not allowed": "Die Art dieser Datei ist nicht erlaubt",
+  "The file is too big": "Die Datei ist zu groß",
+  "Error while uploading file": "Fehler beim Hochladen der Datei",
+  "Error while uploading file(s)": "Fehler beim Hochladen der Datei(en)",
+  "Please wait until the upload ends": "Bitte warten Sie, bis der Upload beendet ist",
+  "Avatar of {{publicName}}": "Avatar von {{publicName}}",
+  "Close": "Schließen",
+  "Restore the original language": "Ursprüngliche Sprache wiederherstellen",
+  "Show translation": "Übersetzung anzeigen",
+  "Translation pending…": "Übersetzung ausstehend...",
```
Comment (gbonaspetti) [inline frontend_lib/i18next.scanner/de/translation.json:139]:
```suggestion
  "Translation pending…": "Übersetzung ausstehend…",
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

**Comment by gbonaspetti** [comment]
Missing translation at frontend/src/component/Search/DateFilter.jsx

File: frontend/src/component/Search/DateFilter.jsx [hunk 4]
```diff
@@ -98,20 +108,36 @@ export class DateFilter extends React.Component {
...
+      pt: 0,
+      de: 1
     }
 
     const FORMAT = {
       en: 'MM/dd/yyyy',
       fr: 'dd/MM/yyyy',
-      pt: 'dd/MM/yyyy'
+      pt: 'dd/MM/yyyy',
+      de: 'TT/MM/JJJJ'
```
Comment (gbonaspetti) [inline frontend/src/component/Search/DateFilter.jsx:140]: I think this part is automatic, not the translation, the place holder translation is in the i18next file, so it should be something like that according to how the date is presented 
```suggestion
      de: 'dd/MM/yyyy'
```

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: backend/tracim_backend/locale/de/LC_MESSAGES/tracim_backend.po [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+"POT-Creation-Date: 2021-05-03 12:31+0200\n"
+"PO-Revision-Date: 2021-04-30 16:42+0100\n"
+"Last-Translator: marie <marie.begue@algoo.fr>\n"
+"Language: de\n"
+"Language-Team: French <kde-i18n-doc@kde.org>\n"
+"Plural-Forms: nplurals=2; plural=(n != 1)\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Generated-By: Babel 2.6.0\n"
```
Comment (inkhey) [inline backend/tracim_backend/locale/de/LC_MESSAGES/tracim_backend.po:21]: Not really important but some value here doesn't look make much sense Language-Team, Report-Msgid-Bugs-To.
but also "ORGANIZATION" in the header.
We should probably add a small issue about configuring theses value properly and take some time just to understand the goal or not of these metadata.

File: frontend/src/util/tinymceInit.js [hunk 6]
```diff
@@ -61,6 +61,8 @@ import { uniqueId } from 'lodash'
           return 'fr_FR'
         case 'pt':
           return 'pt_PT'
+        case 'de':
+          return 'pt_DE'
```
Comment (grignards) [inline frontend/src/util/tinymceInit.js:65]: Shouldn't it be:

```suggestion
          return 'de_DE'
```

**Comment by inkhey** [comment]
updating the file https://github.com/tracim/tracim/blob/b6a8aadc5445b0d204ca8729b3ef24186035af80/frontend/dist/assets/branding.sample/welcome-simple-text.html may also make sense to be explicit about the fact that german is now available.

**Review by gbonaspetti — APPROVED** [review]

**Comment by gbonaspetti** [comment]
Tested on branch

**Pull request merged**